File: solutions/javascript/RomantoInteger/RomantoInteger.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    var a = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000};
    var arr = s.split('');
    var res = 0;
    for (var i=0; i<arr.length; i++) {
        if (i<arr.length-1 && a[arr[i]] < a[arr[i+1]]) {
            res -= a[arr[i]];
        } else {
            res += a[arr[i]];
        }
    }
    return res;
};

File: solutions/java/RomantoInteger/RomantoInteger.java
class Solution {
    public int romanToInt(String s) {
        Map<String, Integer> map = new HashMap<>();
        map.put("I", 1);
        map.put("V", 5);
        map.put("X", 10);
        map.put("L", 50);
        map.put("C", 100);
        map.put("D", 500);
        map.put("M", 1000);
        String[] arr = s.split("");
        int res = 0;
        for (int i = 0; i < arr.length; i++) {
            if (i < arr.length - 1 && map.get(arr[i]) < map.get(arr[i+1])) {
                res -= map.get(arr[i]);
            } else {
                res += map.get(arr[i]);
            }
        }
        return res;
    }
}